src/cpp/utils/IPLocator.cpp

```diff
@@ -88,14 +80,18 @@ bool IPLocator::setIPv4(
         Locator_t& locator,
         const std::string& ipv4)
 {
+    LOCATOR_ADDRESS_INVALID(locator.address);
     //std::string _ipv4 = IPFinder::getIPv4Address(ipv4);
     std::stringstream ss(ipv4);
     int a, b, c, d; //to store the 4 ints
     char ch; //to temporarily store the '.'
 
     if (ss >> a >> ch >> b >> ch >> c >> ch >> d)
     {
-        LOCATOR_ADDRESS_INVALID(locator.address);
+        if (a > 255 || b > 255 || c > 255 || d > 255)
+        {
+            return false;
+        }
```

Comment:
The memory cleaning should only apply to the  last 4 octets, those corresponding to the IPv4 adress